### PR TITLE
(Chore) Implement generic getter for common get API hooks

### PR DESCRIPTION
**Pull request by samuelleeuwenburg**

### CODE REVIEW

File: src/hooks/api/useBuildGetter.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+import useFetch from 'hooks/useFetch'
+import { GetHookResponse } from './types'
+
+export const useBuildGetter = <T, U extends Array<unknown> = Array<any>>(
+  paramBuilder: (...args: U) => [string, Record<string, any>]
```
Comment (MauritsR) [inline src/hooks/api/useBuildGetter.ts:6]: Isn't it better if the second argument were optional?

File: src/hooks/api/useBuildGetter.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+): GetHookResponse<T, U> => {
+  const { data, get: genericGet, isLoading, error, isSuccess } = useFetch<T>()
+  const get = useCallback(
+    (...args: U) => genericGet(...paramBuilder(...args)),
+    []
```
Comment (MauritsR) [inline src/hooks/api/useBuildGetter.ts:11]: missing dependencies

File: src/hooks/api/useBuildGetter.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+import useFetch from 'hooks/useFetch'
+import { GetHookResponse } from './types'
+
+export const useBuildGetter = <T, U extends Array<unknown> = Array<any>>(
+  paramBuilder: (...args: U) => [string, Record<string, any>]
+): GetHookResponse<T, U> => {
+  const { data, get: genericGet, isLoading, error, isSuccess } = useFetch<T>()
```
Comment (MauritsR) [inline src/hooks/api/useBuildGetter.ts:8]: It's not necessary to individually destruct the keys here -> `const { get: genericGet, ...rest} =`

File: src/signals/incident-management/containers/ReporterContainer/useFetchReporter.ts [hunk 4]
```diff
@@ -88,6 +92,15 @@ export const useFetchReporter = (id: string): FetchReporterHook => {
...
   }, [getReporterData])
 
+  useEffect(() => {
+    if (id) {
+      getContextReporter(Number(id), {
+        page: currentPage,
+        pageSize: PAGE_SIZE,
+      })
+    }
+  }, [getContextReporter, id, currentPage, PAGE_SIZE])
```
Comment (MauritsR) [inline src/signals/incident-management/containers/ReporterContainer/useFetchReporter.ts:102]: unnecessary dependency: 'PAGE_SIZE'

**Review by MauritsR — APPROVED** [review]

File: src/hooks/api/useBuildGetter.ts (hunk 2 above)
Comment (samuelleeuwenburg) [reply src/hooks/api/useBuildGetter.ts:11]: this is intentional, to make sure this callback never changes

File: src/hooks/api/useBuildGetter.ts (hunk 3 above)
Comment (samuelleeuwenburg) [reply src/hooks/api/useBuildGetter.ts:8]: 👍

File: src/hooks/api/useBuildGetter.ts (hunk 1 above)
Comment (samuelleeuwenburg) [reply src/hooks/api/useBuildGetter.ts:6]: this does result in some strange syntax, let me know what you think

File: src/signals/incident-management/containers/ReporterContainer/useFetchReporter.ts (hunk 4 above)
Comment (samuelleeuwenburg) [reply src/signals/incident-management/containers/ReporterContainer/useFetchReporter.ts:102]: 👍

File: src/hooks/api/useBuildGetter.ts (hunk 1 above)
Comment (samuelleeuwenburg) [reply src/hooks/api/useBuildGetter.ts]: nevermind, managed to solve it in a cleaner way!

**Pull request merged**